src/Graphics/Datoviz/Types.hs

```diff
@@ -0,0 +1,91 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE ImportQualifiedPost #-}
+{-# LANGUAGE KindSignatures #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+module Graphics.Datoviz.Types where
+
+import Data.Coerce (coerce)
+import Data.Data
+import Data.Vector.Storable qualified as VS
+import Foreign
+import GHC.TypeNats
+
+-- | Fixed-Sized Vector with type `a` and length `b`.
```

Comment:
Personally, I'd prefer to add a dependency on `linear` and use `V2`, `V3` and so on. They have `Storable` instances, and lots of other code built around them. (Graphics is the reason that `linear` was written in the first place, btw.)

Reply:
You're right. Linear is the right call here.